bin/reth/src/args/utils.rs

```diff
@@ -1,37 +1,63 @@
 //! Clap parser utilities
 
-use reth_primitives::{
-    fs, AllGenesisFormats, BlockHashOrNumber, ChainSpec, B256, DEV, GOERLI, HOLESKY, MAINNET,
-    SEPOLIA,
-};
+use reth_primitives::{fs, AllGenesisFormats, BlockHashOrNumber, ChainSpec, B256};
 use std::{
     net::{IpAddr, Ipv4Addr, SocketAddr, ToSocketAddrs},
-    path::PathBuf,
+    path::{Path, PathBuf},
     str::FromStr,
     sync::Arc,
     time::Duration,
 };
+use tracing::{debug, info};
+
+use reth_rpc::{JwtError, JwtSecret};
 
 #[cfg(feature = "optimism")]
 use reth_primitives::{BASE_GOERLI, BASE_MAINNET};
 
+#[cfg(not(feature = "optimism"))]
+use reth_primitives::{DEV, GOERLI, HOLESKY, MAINNET, SEPOLIA};
+
+#[cfg(feature = "optimism")]
+/// Chains supported by op-reth. First value should be used as the default.
+pub const SUPPORTED_CHAINS: &[&str] = &["base", "base_goerli"];
+#[cfg(not(feature = "optimism"))]
+/// Chains supported by reth. First value should be used as the default.
+pub const SUPPORTED_CHAINS: &[&str] = &["mainnet", "sepolia", "goerli", "holesky", "dev"];
+
 /// Helper to parse a [Duration] from seconds
 pub fn parse_duration_from_secs(arg: &str) -> eyre::Result<Duration, std::num::ParseIntError> {
     let seconds = arg.parse()?;
     Ok(Duration::from_secs(seconds))
 }
+/// Attempts to retrieve or create a JWT secret from the specified path.
+
+pub fn get_or_create_jwt_secret_from_path(path: &Path) -> Result<JwtSecret, JwtError> {
+    if path.exists() {
+        debug!(target: "reth::cli", ?path, "Reading JWT auth secret file");
+        JwtSecret::from_file(path)
+    } else {
+        info!(target: "reth::cli", ?path, "Creating JWT auth secret file");
+        JwtSecret::try_create(path)
+    }
+}
```

Comment:
this should be moved to reth bin crate crate

Reply:
> this should be moved to reth bin crate crate

ty lot, we are talking about this file right ? https://github.com/paradigmxyz/reth/blob/main/bin/reth/src/utils.rs

Reply:
yes

Reply:
> yes

ty lot, should be ready !